### PR TITLE
Fix the error "/usr/bin/ld cannot find -lopenMVG_dataio" when not building the voctree module

### PR DESCRIPTION
**Pull request by caymard**
Connected to poparteu/openMVG#112

### CODE REVIEW

**Comment by caymard** [comment]
The build of openMVG with `OpenMVG_BUILD_VOCTREE=ON/OFF` and  `OpenMVG_USE_ALEMBIC=ON/OFF` passed without errors.

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -395,22 +395,20 @@ ENDIF(OpenMVG_USE_OPENCV)
...
-     endif(BOOST_NO_CXX11)
-     include_directories(${Boost_INCLUDE_DIRS})
-     link_directories(${Boost_LIBRARIES})
-     add_definitions(${Boost_DEFINITIONS})
-  ELSE()
-    MESSAGE(STATUS "Boost not found -- disabling the vocabulary tree")
-    UPDATE_CACHE_VARIABLE(OpenMVG_BUILD_VOCTREE OFF)
-  ENDIF(Boost_FOUND)
-ENDIF(OpenMVG_BUILD_VOCTREE)
+OPTION(BOOST_NO_CXX11 "If Boost is not build with C++11 support this must be enabled to avoid symbol conflicts (SCOPED_ENUM)" OFF)
```
Comment (simogasp) [inline src/CMakeLists.txt]: maybe it's better to keep the `IF(OpenMVG_BUILD_VOCTREE)`: if `OpenMVG_BUILD_VOCTREE` is set to FALSE with cmake, it is not necessary to look for boost, isn't it? and Boost is only required, for now, inside the voctree stuff i think.

Comment (caymard) [inline src/CMakeLists.txt]: For the issue poparteu/openMVG#102 we need boost and it is not depending on the voctree.

Comment (simogasp) [inline src/CMakeLists.txt]: Ok

**Pull request merged**